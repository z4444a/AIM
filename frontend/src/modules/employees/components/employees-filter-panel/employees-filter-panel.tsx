import React, { KeyboardEvent, ReactNode } from 'react';
import TextField from '@material-ui/core/TextField';
import { Style } from './styles';
import { CustomFilterCmpContext } from '../../../../commons/components/aim-data-grid/aim-data-grid';
import { WithTranslation } from 'react-i18next';

export enum FilterField {
  EMPLOYEE_LAST_NAME = 'lastName',
}

export interface Props {
  context: CustomFilterCmpContext;
}

export type InternalProps = Props & Style & WithTranslation;

export class EmployeesFilterPanel extends React.PureComponent<InternalProps> {
  public render(): ReactNode {
    const { classes, context, t } = this.props;
    const name =
      context.values && context.values[FilterField.EMPLOYEE_LAST_NAME]
        ? context.values[FilterField.EMPLOYEE_LAST_NAME] + ''
        : '';
    return (
      <div className={classes.container}>
        <TextField
          value={name}
          fullWidth
          onChange={this.handleFilterTypeChange}
          onKeyPress={this.handleKeyPress}
          label={t('employeePage.grid.columns.lastName')}
        />
      </div>
    );
  }

  private handleKeyPress = (event: KeyboardEvent) => {
    const { context } = this.props;
    if (event.key === 'Enter' && typeof context.applyFilters === 'function') {
      context.applyFilters();
      event.preventDefault();
    }
  };

  private handleFilterTypeChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { context } = this.props;
    const values = Object.assign({}, context.values, {
      [FilterField.EMPLOYEE_LAST_NAME]: event.target.value,
    });
    context.changeFilters(values);
  };
}
