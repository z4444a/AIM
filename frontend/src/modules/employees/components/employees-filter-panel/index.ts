import withStyles from '@material-ui/core/styles/withStyles';
import { styles } from './styles';
import { withTranslation } from 'react-i18next';
import { EmployeesFilterPanel } from './employees-filter-panel';

export default withStyles(styles)(withTranslation('common')(EmployeesFilterPanel));
