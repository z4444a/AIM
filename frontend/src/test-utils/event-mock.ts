export const getMockChangeEvent = (value: string) => {
  return {
    target: {
      value: value,
    },
  };
};

export const getMockKeyPressEvent = (key: string) => {
  return {
    key: key,
    preventDefault: jest.fn().mockName('preventDefault'),
  };
};
