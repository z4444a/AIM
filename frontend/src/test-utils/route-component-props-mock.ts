import { RouteComponentProps } from 'react-router';
import { Href } from 'history';

export function getMockRouterProps<P>(data: P): RouteComponentProps {
  const location = {
    hash: '',
    key: '',
    pathname: '',
    search: '',
    state: {},
  };

  const props: RouteComponentProps<P> = {
    match: {
      isExact: true,
      params: data,
      path: '',
      url: '',
    },
    location: location,
    history: {
      length: 2,
      action: 'POP',
      location: location,
      push: () => {},
      replace: () => {},
      go: () => {},
      goBack: () => {},
      goForward: () => {},
      block: () => () => {},
      createHref: () => {
        const temp: Href = '';
        return temp;
      },
      listen: () => () => {},
    },
    staticContext: {},
  };

  return props;
}
