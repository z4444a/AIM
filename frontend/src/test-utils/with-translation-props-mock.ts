import { WithTranslation } from 'react-i18next';

export function getMockWithTranslationProps(): WithTranslation {
  // eslint-disable-next-line
  const t = ((key: string) => `${key}_i18n`) as any;
  // eslint-disable-next-line
  const i18n = {} as any;
  const props: WithTranslation = {
    tReady: true,
    t,
    i18n,
  };

  return props;
}
