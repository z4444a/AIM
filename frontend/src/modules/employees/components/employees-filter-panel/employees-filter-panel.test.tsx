import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import { EmployeesFilterPanel, InternalProps } from './employees-filter-panel';
import { getMockWithTranslationProps } from '../../../../test-utils/with-translation-props-mock';

describe('EmployeesFilterPanel', () => {
  function getComponentProps(): InternalProps {
    const withTranslation = getMockWithTranslationProps();
    const props: InternalProps = {
      classes: {
        container: 'container',
      },
      ...withTranslation,
      context: {
        values: {},
        changeFilters: () => {},
        applyFilters: () => {},
      },
    };
    return props;
  }

  function generateShallowWrapper() {
    return shallow(<EmployeesFilterPanel {...getComponentProps()} />);
  }

  it('should render correctly', () => {
    const wrapper = generateShallowWrapper();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
