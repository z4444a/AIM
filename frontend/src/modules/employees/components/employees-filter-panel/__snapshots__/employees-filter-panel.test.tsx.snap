// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeesFilterPanel should render correctly 1`] = `
<div
  className="container"
>
  <TextField
    fullWidth={true}
    label="employeePage.grid.columns.lastName_i18n"
    onChange={[Function]}
    onKeyPress={[Function]}
    required={false}
    select={false}
    value=""
    variant="standard"
  />
</div>
`;
